src/components/Home/Home.tsx

```diff
@@ -23,28 +38,68 @@ export function Home() {
           <li>Insert other eligbility rules</li>
         </ul>
       </Banner>
-      <br />
-      <br />
     </>
   ) : null;
+
+  const handleSubmit = () => {
+    const postalCodeRegex = /[a-zA-Z][0-9][a-zA-Z](-| |)[0-9][a-zA-Z][0-9]/;
+    if (postalCodeRegex.test(postalCode)) {
+      setShouldRenderPharmacyList(true);
+      setShouldShowPostalPrompt(false);
+    } else {
+      setShouldShowInvalidPostal(true);
+    }
+  };
+  const afterPostalEntryMarkup = shouldRenderPharmacyList ? (
+    <>
+      <Layout.Section>{dismissableBannerMarkup}</Layout.Section>
+      <PharmacyList />
+    </>
+  ) : null;
+
+  const invalidPostalCodeMessage = shouldShowInvalidPostal
+    ? "You have entered an invalid postal code"
+    : undefined;
+
+  const promptForPostalMarkup = shouldShowPostalPrompt ? (
+    <>
+      <Card>
+        <Card.Section>
+          <Form onSubmit={handleSubmit}>
+            <FormLayout>
+              <TextField
+                value={postalCode}
+                onChange={(postal) => setPostalCode(postal)}
+                label="Please enter your postal code (Example: K2T 0E5):"
+                helpText={
+                  <span>
+                    We’ll use this postal code to find the closest available
+                    vaccines.
+                  </span>
+                }
+                error={invalidPostalCodeMessage}
```

Comment:
@picuron By virtue of adding a message to the error prop, `TextField` will take care of showing an inline error and setting all the aria properties.